### PR TITLE
gpu: Fix shaders in compute pipelines

### PR DESCRIPTION
**Pull request by Tony-LunarG**
Replace instrumented shaders in compute pipelines when descriptor set is unavailable.  Mirrors treatment of graphics pipelines

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5141,37 +5141,54 @@ void CoreChecks::PostCallRecordCreateGraphicsPipelines(VkDevice device, VkPipeli
 bool CoreChecks::PreCallValidateCreateComputePipelines(VkDevice device, VkPipelineCache pipelineCache, uint32_t count,
                                                        const VkComputePipelineCreateInfo *pCreateInfos,
                                                        const VkAllocationCallbacks *pAllocator, VkPipeline *pPipelines,
-                                                       void *pipe_state_data) {
+                                                       void *ccpl_state_data) {
     bool skip = false;
-    std::vector<std::unique_ptr<PIPELINE_STATE>> *pipe_state =
-        reinterpret_cast<std::vector<std::unique_ptr<PIPELINE_STATE>> *>(pipe_state_data);
-    pipe_state->reserve(count);
+    create_compute_pipeline_api_state *ccpl_state = reinterpret_cast<create_compute_pipeline_api_state *>(ccpl_state_data);
```
[the remark this answers is not in the document]
Comment (Tony-LunarG) [reply layers/core_validation.cpp]: Auto done

File: scripts/layer_chassis_generator.py [hunk 2]
```diff
@@ -1124,25 +1124,28 @@ class ValidationObject {
     auto layer_data = GetLayerDataPtr(get_dispatch_key(device), layer_data_map);
     bool skip = false;
 
-#ifndef BUILD_CORE_VALIDATION
-    struct PIPELINE_STATE {};
+#ifdef BUILD_CORE_VALIDATION
+    create_compute_pipeline_api_state ccpl_state{};
+#else
+    struct create_compute_pipeline_api_state {
```
[the remark this answers is not in the document]
Comment (Tony-LunarG) [reply scripts/layer_chassis_generator.py]: VkResult result = DispatchCreateComputePipelines(device, pipelineCache, createInfoCount, ccpl_state.pCreateInfos, pAllocator, pPipelines);

is generated in chassis.cpp.  There needs to be a pCreateInfos.

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -5141,37 +5141,54 @@ void CoreChecks::PostCallRecordCreateGraphicsPipelines(VkDevice device, VkPipeli
...
+
+    ccpl_state->pipe_state.reserve(count);
     for (uint32_t i = 0; i < count; i++) {
         // Create and initialize internal tracking data structure
-        pipe_state->push_back(unique_ptr<PIPELINE_STATE>(new PIPELINE_STATE));
-        (*pipe_state)[i]->initComputePipeline(&pCreateInfos[i]);
-        (*pipe_state)[i]->pipeline_layout = *GetPipelineLayout(pCreateInfos[i].layout);
+        ccpl_state->pipe_state.push_back(unique_ptr<PIPELINE_STATE>(new PIPELINE_STATE));
+        (ccpl_state->pipe_state)[i]->initComputePipeline(&pCreateInfos[i]);
```
[the remark this answers is not in the document]
Comment (Tony-LunarG) [reply layers/core_validation.cpp]: Done

File: layers/core_validation.cpp [hunk 4]
```diff
@@ -5115,37 +5115,54 @@ void CoreChecks::PostCallRecordCreateGraphicsPipelines(VkDevice device, VkPipeli
...
-        pipe_state->push_back(unique_ptr<PIPELINE_STATE>(new PIPELINE_STATE));
-        (*pipe_state)[i]->initComputePipeline(&pCreateInfos[i]);
-        (*pipe_state)[i]->pipeline_layout = *GetPipelineLayout(pCreateInfos[i].layout);
+        ccpl_state->pipe_state.push_back(unique_ptr<PIPELINE_STATE>(new PIPELINE_STATE));
+        ccpl_state->pipe_state.back()->initComputePipeline(&pCreateInfos[i]);
+        ccpl_state->pipe_state.back()->pipeline_layout = *GetPipelineLayout(pCreateInfos[i].layout);
 
         // TODO: Add Compute Pipeline Verification
-        skip |= ValidateComputePipeline((*pipe_state)[i].get());
+        skip |= ValidateComputePipeline(ccpl_state->pipe_state[i].get());
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: also .back() to be consistent :)

Comment (Tony-LunarG) [reply layers/core_validation.cpp]: Done

**Pull request merged**